From 757a1bbf3a3da2cf8c713650e59e0a90349b7dc3 Mon Sep 17 00:00:00 2001
From: 24bit192khz <118996408+24bit192kHz@users.noreply.github.com>
Date: Mon, 28 Sep 2026 19:44:58 +0300
Subject: [PATCH 08/14] coreboot: Restore the original preloaded module list

Dropping ahci/pata/ehci/uhci/ohci moved the NVMe controller init ~56 ms
earlier. On the next boots the first NVMe read took 1.59 s instead of
7 ms, and a warm reboot hung with no output; the lbmk NVMe driver has its
CSTS.RDY and completion timeouts commented out, so it waits forever.
Keep the old timing until that driver gets real timeouts; the single
etc/grub.cfg read from the previous commit stays.
---
 Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index a010c4502..2c86dbbf6 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -434,7 +434,7 @@ if COND_i386_coreboot
 FS_PAYLOAD_MODULES ?= $(shell cat grub-core/fs.lst)
 default_payload.elf: grub-mkstandalone grub-mkimage FORCE
 	test -f $@ && rm $@ || true
-	pkgdatadir=. ./grub-mkstandalone --grub-mkimage=./grub-mkimage -O i386-coreboot -o $@ --modules='nvme part_msdos ext2 fat at_keyboard part_gpt cbfs' --install-modules='ls linux search configfile normal cbtime cbls memrw iorw minicmd lsmmap lspci halt reboot hexdump pcidump regexp setpci lsacpi chain test serial multiboot cbmemc linux16 gzio echo help syslinuxcfg xnu $(FS_PAYLOAD_MODULES) password_pbkdf2 $(EXTRA_PAYLOAD_MODULES)' --fonts= --themes= --locales= -d grub-core/ /boot/grub/grub.cfg=$(srcdir)/coreboot.cfg
+	pkgdatadir=. ./grub-mkstandalone --grub-mkimage=./grub-mkimage -O i386-coreboot -o $@ --modules='ahci pata nvme ehci uhci ohci usb_keyboard usbms part_msdos ext2 fat at_keyboard part_gpt usbserial_usbdebug cbfs' --install-modules='ls linux search configfile normal cbtime cbls memrw iorw minicmd lsmmap lspci halt reboot hexdump pcidump regexp setpci lsacpi chain test serial multiboot cbmemc linux16 gzio echo help syslinuxcfg xnu $(FS_PAYLOAD_MODULES) password_pbkdf2 $(EXTRA_PAYLOAD_MODULES)' --fonts= --themes= --locales= -d grub-core/ /boot/grub/grub.cfg=$(srcdir)/coreboot.cfg
 endif
 
 endif
-- 
2.55.0

